unstructured/ingest/connector/delta_table.py

```diff
@@ -0,0 +1,124 @@
+import logging
+import os
+import uuid
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Any, Dict, List, Optional, Union
+
+import pandas
+import pyarrow
+from deltalake import DeltaTable
+from pyarrow import RecordBatch
+
+from unstructured.ingest.interfaces import (
+    BaseConnector,
+    BaseConnectorConfig,
+    BaseIngestDoc,
+    ConnectorCleanupMixin,
+    IngestDocCleanupMixin,
+    StandardConnectorConfig,
+)
+from unstructured.ingest.logger import make_default_logger
+from unstructured.utils import requires_dependencies
```

Comment:
something is not quite right with order of imports. when I run the test prior to installing the extras I get an error specifically about importing pyarrow rather than the [require_dependencies warning message](https://github.com/Unstructured-IO/unstructured/blob/59962bc6d50b0ed6f8806f42d79767b5c4563e79/unstructured/utils.py#L40-L41) to install the appropriate extra., probably need to move imports and decorate with extra arg.

Reply:
@rbiseck3 bump ^ probably need to move the top level imports inside relevant classes/functions, otherwise one just gets the error:

```
from deltalake import DeltaTable
ModuleNotFoundError: No module named 'deltalake'
Error: No module named 'deltalake'
```